### ISSUE TITLE
Ignore default routes from dhcp

### ISSUE THREAD

**Issue by resetsa** [k8snetworkplumbingwg/multus-cni#748]
Hi all!
Help me. Please describe me, how i can ignore default routes from dhcp server.
At this moment i'm have 2 default routes in pod.
From dhcp + k8s cluster default.
```
/ # ip route 
default via 192.168.113.1 dev vlan10
default via 169.254.1.1 dev eth0
169.254.1.1 dev eth0 scope link
192.168.113.0/28 dev vlan10 scope link  src 192.168.113.7
```

pod.yaml
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: alpine
  namespace: ns-test
  annotations:
    k8s.v1.cni.cncf.io/networks: |
      [
         { "name":"vlan10", "interface":"vlan10","mac":"B2:7F:B2:F6:14:5A" }
      ]
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
          - matchExpressions:
            - key: powersa.io/vlan10
              operator: Exists
  containers:
  - image: alpine:3.12.8
    command:
      - /bin/sh
      - "-c"
      - "sleep 60m"
    imagePullPolicy: IfNotPresent
    name: alpine
    resources:
      limits:
        cpu: "0.2"
        memory: "128Mi"
      requests:
        cpu: "0.1"
        memory: "64Mi"
  restartPolicy: Always
```

vlan10.yaml
```yaml
apiVersion: "k8s.cni.cncf.io/v1"
kind: NetworkAttachmentDefinition
metadata:
  name: vlan10
  namespace: ns-test
spec:
  config: '{
      "cniVersion": "0.3.1",
      "type": "macvlan",
      "master": "vlan10",
      "ipam": {
        "type": "dhcp"
      }
    }'
```

**Comment by dougbtv** [comment]
One thing to try is the route-override CNI plugin, which you can chain: https://github.com/redhat-nfvpe/cni-route-override

Or... manually or otherwise change the routing.

Having two default routes seems.... interesting, not used to seeing that.

**Comment by MaxThom** [comment]
having the same issue here

**Comment by n0rad** [comment]
```
    {
      "cniVersion": "0.3.0",
      "name": "mymacvlan",
      "plugins": [
        {
          "type": "macvlan",
          "master": "SOMETHING",
          "mode": "bridge",
          "ipam": {
            "type": "dhcp",
            "request": [
              {
                "skipDefault": true,
                "option": "subnet-mask"
              }
            ]
          }
        }
      ]
    }

```

**Comment by virtualdxs** [comment]
Can this be added as an example in the documentation?

**Comment by n0rad** [comment]
It's kind of already documented [here](https://www.cni.dev/plugins/current/ipam/dhcp/#example-configuration), but not super explicit when you are not used to dhcp requests.

I understood it when reading [this part](https://github.com/containernetworking/plugins/blob/fec2d62676cbe4f2fd587b4840c7fc021bead3f9/plugins/ipam/dhcp/lease.go#L85-L88) of the code

**Comment by lgo** [comment]
Amazing, thank you for the solution! I was also browing through the plugins (including the DHCP IPAM) but didn't quite spot that - you're right that the option isn't quite obvious.

Like the OP, I discovered I had two default routes while trying to debug issues which led me here. With a fairly plain `k3s` deployment, my DNS resolution broke once I added a configuration like the OPs (`macvlan` with `dhcp`, albeit I was also trying `host-local` and unsuspectedly copied `0.0.0.0/0` routes)

I'm mostly dropping this with a few more pointers (like k3s) for anyone in the future searching around for symptoms.
